File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        long answer = 0;
        long[] dp = new long[n+1];
        
        if(n <= 2){
            return n;
        }
        long a = 1L, b = 2L;
        for(int i=3;i<=n;i++){
            answer = (a + b)%1234567;
            a = b;
            b = answer;
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.kt
class Solution {
    fun solution(n: Int): Long {
        var answer: Long = 0
        var dp = LongArray(n+1) {0}
        if(n == 1) return 1
        dp[1] = 1
        dp[2] = 2
        
        for(index in 3..n){
            dp[index] = (dp[index-1] + dp[index-2])%1234567
        }
        
        return dp[n]
    }
}